src/contracts-registry/contracts-registry.contract.ts

```diff
@@ -0,0 +1,76 @@
+import {
+  ContractsRegistryAction,
+  ContractsRegistryCurrentTxIdData,
+  ContractsRegistryRegisterContractsData,
+  ContractsRegistryResult,
+  ContractsRegistryState, VersionData
+} from "./types";
+import {Tools} from "../common/Tools";
+import {ContractAdmin} from "../common/ContractAdmin";
+
+declare type ContractResult = { state: ContractsRegistryState } | { result: ContractsRegistryResult }
+declare const ContractError: any;
+declare const SmartWeave: any;
+
+export function handle(state: ContractsRegistryState, action: ContractsRegistryAction): ContractResult {
+  const input = action.input;
+
+  ContractAdmin.checkAndCreate(
+    action.caller,
+    input.function,
+    state.contractAdmins,
+    ["registerContracts"]);
+
+  switch (input.function) {
+    case "registerContracts":
+      const registerContractsData = input.data as ContractsRegistryRegisterContractsData;
+      Tools.initIfUndefined(state, "versions", {});
+      const registerVersion = registerContractsData.version || getLatestVersion(state.versions);
```

Comment:
What if the version is not in the "v...." format?

Reply:
I've added validation and tests.